src/main/java/ui/components/pickers/MilestonePickerDialog.java

```diff
@@ -99,9 +104,9 @@ private void setupButtons(DialogPane milestonePickerDialogPane) {
     private void setConfirmResultConverter() {
         setResultConverter((dialogButton) -> {
             List<PickerMilestone> finalList = state.getCurrentMilestonesList();
-            if (hasSelectedMilestone(finalList)) {
-                return new MilestonePickerDialogResponse(dialogButton,
-                                                         Optional.of(getSelectedMilestone(finalList).get().getId()));
+            Optional<PickerMilestone> selectedMilestone = PickerMilestone.getSelectedMilestone(finalList);
+            if (selectedMilestone.isPresent()) {
+                return new MilestonePickerDialogResponse(dialogButton, Optional.of(selectedMilestone.get().getId()));
```

Comment:
Can we use `selectedMilestone.map(PickerMilestone::getId)` here? Then I think we can get rid of the `Optional.of()`.